special-pages/pages/duckplayer/app/features/playback-events.js

```diff
@@ -0,0 +1,85 @@
+import { EVENT_PLAY, EVENT_PAUSE } from '../components/PlaybackControl.jsx';
+
+/**
+ * @typedef {import("./iframe").IframeFeature} IframeFeature
+ */
+
+/**
+ * Exposes video element in iframe.
+ *
+ * @implements IframeFeature
+ */
+export class PlaybackEvents {
+    /** @type {MutationObserver} */
+    observer;
+
+    /** @type {[EVENT_PLAY|EVENT_PAUSE, (() => void)][]} */
+    handlers = [];
+
+    destroy() {
+        if (this.observer) this.observer.disconnect();
+
+        this.handlers.forEach(([event, handler]) => {
+            window.removeEventListener(event, handler);
+        });
+    }
+
+    /**
+     * @param {HTMLIFrameElement} iframe
+     */
+    iframeDidLoad(iframe) {
+        const documentBody = iframe.contentWindow?.document?.body;
+
+        if (documentBody) {
+            const videoElement = documentBody.querySelector('video');
+
+            // Check if iframe already contains video
+            if (videoElement) {
+                this.addHandlersToVideo(videoElement);
+            } else {
+                // No video found. Observe iframe's document for changes
+                this.observer = new MutationObserver(this.handleMutation.bind(this));
+
+                this.observer.observe(documentBody, {
+                    childList: true,
+                    subtree: true, // Observe all descendants of the body
+                });
+            }
+        }
+
+        return () => {
+            this.destroy();
+        };
+    }
```

Comment:
Actually, why not just look up the video element on click? then you don’t need to find it early, or ave any mutation observers. Just register your event handlers, and when you want to pause the video, just look it up with querySelector

Reply:
Of course!!! 🤦